rbac/internal/middleware.py

```diff
@@ -47,14 +49,17 @@ def process_request(self, request):
         except (JSONDecodeError, binascii.Error):
             logger.exception("Invalid X-RH-Identity header.")
             return HttpResponseForbidden()
-
-        user = User()
+        user = build_internal_user(request, json_rh_auth)
         try:
             if not json_rh_auth["identity"]["type"] == "Associate":
                 return HttpResponseForbidden()
             user.username = json_rh_auth["identity"]["associate"]["email"]
             user.admin = True
```

Comment:
It looks like some of this logic is duplicated now in `build_internal_user`. Maybe we can remove it from here?

Reply:
Yeah, there was a bit of duplication going on there, I'll look back over it